File: src/Web/JavaScriptJS/patterns/presentation_abstraction_control.js
const assert = require("node:assert/strict");
/** Coordinates a PAC agent whose control mediates presentation and abstraction. */
function run() {
  const abstraction = { value: 2 };
  const presentation = { text: "" };
  const control = { refresh() { presentation.text = `value=${abstraction.value}`; } };
  control.refresh();
  assert.equal(presentation.text, "value=2");
  return "presentation-abstraction-control";
}
module.exports = { run };
if (require.main === module) run();

File: src/Functional/Elixir/patterns/presentation_abstraction_control.exs
control = fn abstraction, delta ->
  Map.update!(abstraction, :value, &(&1 + delta))
end

presentation = fn abstraction -> to_string(abstraction.value) end
updated = control.(%{value: 1}, 2)

unless presentation.(updated) == "3" do
  raise "PAC"
end
